lib/screens/event/event.dart

```diff
@@ -46,55 +45,69 @@ class _EventPageState extends State<EventPage> {
+    locator<EventService>().getEvent(widget.eventId).then((value) => setState(() {
+          this.event = value;
+          if (this.event != null) {
+            this.drinkPackageAnswer = "Alkohol";
+            this.defaultGroup = event!.groups![0];
+            this.group = defaultGroup;
+            if (this.group == null) {
+              this.displayGroupInput = true;
+            }
+          }
+        }));
     locator<UserService>().getUser().then((value) => setState(() {
           this.foodPreferences['en'] = [...(value.food_preferences ?? [])];
           this.foodPreferences['sv'] = [...(value.food_preferences ?? [])];
           this.foodCustom = value.food_custom;
           for (int i = 0; i < (this.foodPreferences['sv']?.length ?? 0); i++) {
-            this.foodPreferences['sv']![i] =
-                foodPrefsDisplay[this.foodPreferences['sv']![i]] ?? "";
+            this.foodPreferences['sv']![i] = foodPrefsDisplay[this.foodPreferences['sv']![i]] ?? "";
           }
         }));
-
     super.initState();
   }
 
   void update() {
-    locator<EventService>()
-        .getEvent(widget.eventId)
-        .then((value) => setState(() {
-              this.event = value;
+    locator<EventService>().getEvent(widget.eventId).then((value) => setState(() {
+          this.event = value;
+          if (event != null) {
+            if (event!.user_types != null) {
+              this.userType = event!.user_types![0][0];
+            } else {
               this.userType = null;
-              this.group = null;
-              this.answer = null;
-            }));
+            }
+            // when we update (basically undo signup) make sure to go back to normal defaults to avoid weird behaviour
+            this.group = event!.groups![0];
+          } else {
+            this.group = null;
+            this.userType = null;
+          }
+          this.answer = null;
+        }));
     locator<UserService>().getUser().then((value) => setState(() {
           this.foodPreferences['en'] = [...(value.food_preferences ?? [])];
           this.foodPreferences['sv'] = [...(value.food_preferences ?? [])];
           this.foodCustom = value.food_custom;
           for (int i = 0; i < (this.foodPreferences['sv']?.length ?? 0); i++) {
-            this.foodPreferences['sv']![i] =
-                foodPrefsDisplay[this.foodPreferences['sv']![i]] ?? "";
+            this.foodPreferences['sv']![i] = foodPrefsDisplay[this.foodPreferences['sv']![i]] ?? "";
           }
         }));
   }
 
   void sendSignup() async {
-    EventUser eventUser =
-        EventUser(answer, group?.id, customGroup, userType, drinkPackageAnswer);
+    EventUser eventUser = EventUser(answer, group?.id, customGroup, userType, drinkPackageAnswer);
     /* just to be sure */
     if (group?.id != null) {
       customGroup = null;
     }
+    if (drinkPackageAnswer == null || drinkPackageAnswer == "") {
+      setState(() {
+        drinkPackageAnswer = "Inget";
```

Comment:
Make this a const String so that we know that it matches with the value one sets on row 506? (Same would apply to "Alkohol" on row 51)